server/src/main/java/org/elasticsearch/action/admin/cluster/node/tasks/get/TransportGetTaskAction.java

```diff
@@ -210,7 +212,7 @@ void getFinishedTaskFromIndex(Task thisTask, GetTaskRequest request, ActionListe
         GetRequest get = new GetRequest(TaskResultsService.TASK_INDEX, TaskResultsService.TASK_TYPE,
                 request.getTaskId().toString());
         get.setParentTask(clusterService.localNode().getId(), thisTask.getId());
-        client.get(get, new ActionListener<GetResponse>() {
+        new OriginSettingClient(client, TASKS_ORIGIN).get(get, new ActionListener<GetResponse>() {
```

Comment:
Any reason why this is not done once in the TransportGetTaskAction's constructor?

Reply:
I had a good reason for doing it this way when I wrote it at first but now it doesn't seem like a good reason. I'm switching it to what you suggest.

Reply:
:+1: